staging/src/k8s.io/apiserver/pkg/endpoints/metrics/metrics.go

```diff
@@ -61,6 +61,22 @@ const (
  * the metric stability policy.
  */
 var (
+	deprecatedRegisteredWatchers = compbasemetrics.NewGaugeVec(
+		&compbasemetrics.GaugeOpts{
+			Name:           "deprecated_apiserver_registered_watchers",
+			Help:           "Number of currently registered watchers for a given resources, and it will be deprecated soon.",
```

Comment:
Actually you don't need to do this, we have machinery that will inject Deprecation into the help text.

In order to leverage that though you need to do something like this:

https://github.com/kubernetes/kubernetes/blob/4d78db54a58e250b049c4fe17ac484e5c3ec662d/pkg/kubelet/metrics/metrics.go#L370

Reply:
Ah, that's great. Updated

Reply:
(FYI it should be 1.23, since the branch for 1.22 is closed currently)